uxarray/core/api.py

```diff
@@ -261,3 +262,27 @@ def open_mfdataset(grid_filename_or_obj: str,
     uxds = UxDataset(ds, uxgrid=uxgrid, source_datasets=str(paths))
 
     return uxds
+
+
+def enable_jit_cache(enable_jit_cache):
+    """Allows Numba's JIT cache to be turned on and off.
+
+    This cache variable lets @njit cache the machine code generated
+    between runs, allowing for faster run times due to the fact that the
+    code doesn't need to regenerate the machine code every run time. Our
+    use case here was to study performance, in regular usage one might
+    never turn off caching as it will only help if frequently modifying
+    the code or because users have very limited disk space. Default is
+    on
+    """
+    uxarray.utils.constants.ENABLE_JIT_CACHE = enable_jit_cache
+
+
+def enable_jit(enable_jit):
+    """Allows Numba's JIT application to be turned on and off.
+
+    This lets users choose whether they want machine code to be
+    generated to speed up the performance of the code on large files.
+    Default is on
+    """
+    uxarray.utils.constants.ENABLE_JIT = enable_jit
```

Comment:
I'd consider a slight design change to these functions, since we'd like to avoid needing to pass through a parameter since there are no protections in place (user could pass through anything). The following would be a safer approach. 

```Python
def enable_jit_cache():
    uxarray.utils.constants.ENABLE_JIT_CACHE = True
```
```Python
def disable_jit_cache():
    uxarray.utils.constants.ENABLE_JIT_CACHE = False
```

```Python
def enable_jit():
    uxarray.utils.constants.ENABLE_JIT = True
```

```Python
def disable_jit():
    uxarray.utils.constants.ENABLE_JIT = False
```

Reply:
That is a good idea